Application/Services/CategoryService.cs

```diff
@@ -1,15 +1,18 @@
 ﻿using Application.Dtos;
 using Application.Interfaces;
+using Application.Validators;
 using Domain.Models;
+using FluentValidation;
 using Infrastructure.Interfaces;
 using Mapster;
 
 namespace Application.Services;
 
-public class CategoryService(ICategoryRepository categoryRepository) : ICategoryService
+public class CategoryService(ICategoryRepository categoryRepository, IValidator<CategoryDto> categoryValidator) : ICategoryService
 {
     public async Task<CategoryResponseDto> CreateCategoryAsync(CategoryDto categoryDto)
     {
+        categoryValidator.ValidateAndThrow(categoryDto);
```

Comment:
I dont see the same for Product Service (thats the main point, where and when we validate our request - when you call the validator)